src/NLog/Internal/ObjectPools/ReusableObjectCreator.cs

```diff
@@ -41,16 +41,18 @@ namespace NLog.Internal
     internal class ReusableObjectCreator<T> where T : class
     {
         protected T _reusableObject;
-        private readonly Action<T> _clearObject;
-        private readonly Func<int, T> _createObject;
+        private readonly Action<int, int, T> _clearObject;
+        private readonly Func<int, int, T> _createObject;
         private readonly int _initialCapacity;
+        private readonly int _maxCapacity;
 
-        protected ReusableObjectCreator(int initialCapacity, Func<int, T> createObject, Action<T> clearObject)
+        protected ReusableObjectCreator(int initialCapacity, int maxCapacity, Func<int, int, T> createObject, Action<int, int, T> clearObject)
         {
-            _reusableObject = createObject(initialCapacity);
+            _reusableObject = createObject(initialCapacity, maxCapacity);
             _clearObject = clearObject;
             _createObject = createObject;
             _initialCapacity = initialCapacity;
+            _maxCapacity = maxCapacity;
```

Comment:
Maybe logic about handling `_initialCapacity` + `_maxCapacity` should just be kept inside the delegate-methods `createObject` and `clearObject`.

Thus the base-class `ReusableObjectCreator` doesn't have `_initialCapacity` + `_maxCapacity` as members.